scripts/find_packages.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+import os.path
+import sys
+
+scripts_path = os.path.dirname(os.path.realpath(__file__))
+bb_lib_path = os.path.abspath(scripts_path + '/../../poky/bitbake/lib')
+sys.path = sys.path + [bb_lib_path]
+
+import bb.fetch2
+import bb.tinfoil
+
+
+def print_deps(tinfoil, abcd_file, rn):
+    try:
+        info = tinfoil.get_recipe_info(rn)
+    except:
+        # fails on hostperl-runtime-native, virtual/libintl-native, grep-native, virtual/libiconv-native
+        print('Failed to get recipe info for: %s' % rn)
+        return []
+    if not info:
+        # fails on the above and virtual/crypt-native
+        print('No recipe info found for: %s' % rn)
+        return []
+    append_files = tinfoil.get_file_appends(info.fn)
+    appends = True
+    data = tinfoil.parse_recipe_file(info.fn, appends, append_files)
+    src_uri = data.getVar('SRC_URI').split()
+    lic = data.getVar('LICENSE')
+    summary = data.getVar('SUMMARY')
+    description = data.getVar('DESCRIPTION')
+    homepage = data.getVar('HOMEPAGE')
+    srcrev = data.getVar('SRCREV')
+    branch = data.getVar('BRANCH')
+    depends = data.getVar('DEPENDS').split()
+
+    abcd_file.write('- id:\n')
+    abcd_file.write('    package_manager: "Yocto"\n')
+    abcd_file.write('    name: "%s"\n' % info.pn)
+    abcd_file.write('    version: "%s"\n' % info.pv)
+    abcd_file.write('  declared_lics:\n')
+    abcd_file.write('  - "%s"\n' % lic)
+    if summary:
+        abcd_file.write('  description: "%s"\n' % summary)
+    else:
+        abcd_file.write('  description: "%s"\n' % description)
+    abcd_file.write('  homepage_url: "%s"\n' % homepage)
+    abcd_file.write('  source_artifact:\n')
+    repos = []
+    for src in src_uri:
+        # Strip options.
+        # TODO: ignore files with apply=false?
+        semi_pos = src.find(';')
+        if semi_pos > 0:
+            src = src[:semi_pos]
+        if src[0:7] == 'file://':
```

Comment:
All of these can be be replaced by `.startswith()`

Reply:
Or actually, it could first extract the protocol with a split (other comment) and only filter on this part afterwards.